frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -275,7 +271,22 @@ fn construct_runtime_final_expansion(
 	let integrity_test = decl_integrity_test(&scrate);
 	let static_assertions = decl_static_assertions(&name, &pallets, &scrate);
 
+	let warning = if let Some(where_section) = where_section {
+		Some(
+			proc_macro_warning::Warning::new_deprecated("WhereSection")
+				.old("use where section")
+				.new("use `frame_system::Config` to set the `Block` type and remove this section")
+				.help_links(&["https://github.com/paritytech/substrate/pull/14193"])
+				.span(where_section.span)
+				.build(),
```

Comment:
This is just a thought and not a request to change anything. Currently we aim for a soft deprecation before a hard deprecation with a warning to give time to the docs team to update documentation.
But this has proven to be very slow, so there's a new proposal to go towards your approach https://github.com/paritytech/substrate/issues/13976#issuecomment-1591225338